### ISSUE TITLE
Proposal to Handle VOMS-Compatible Groups with Slashes

### ISSUE THREAD

**Issue by Panos512** [indigo-iam/iam#886]
Dear IAM Developers,

CMS has a request I would like to discuss with you. I will explain the issue here and outline a few possible approaches to address it.

The issue concerns how VOMS-compatible services consume groups from proxy certificates. These services follow the pattern:
`{vo_name}/{group}/Role={role}/Capability={capability}`

The problem arises with IAM groups that include slashes in their names, such as /cms/country and /cms/country/de. VOMS services interpret the following:
```
/cms/country/Role=NULL/Capability=NULL  
/cms/country/de/Role=NULL/Capability=NULL  
```
as two entries for the same "country" group. Unfortunately, modifying how these services parse groups is impractical because there are too many of them, and their codebases are complex and difficult to update.

For this specific use case, CMS could populate new groups without slashes (e.g., `/cms/de`, `/cms/it`, `/cms/us`, etc.). However, I believe there may be a more generic and elegant solution.

My suggestion is to introduce a VOMS-Name attribute for groups. If this attribute is set, it would determine how the group is exposed in the proxy certificate. If it is not set, the group would not appear in the certificate at all. This approach would allow us to control both how groups are represented in certificates and which groups are included.

Does this solution sound reasonable to you?

**Comment by giacomini** [comment]
Sorry, I don't understand the issue. The groups `/cms/country` and `/cms/country/de` **are** different groups, there is no `/` in the middle of a group name. The `/` character is forbidden in a group name.

**Comment by Panos512** [comment]
I believe `wlcg.optional-group` only affects the groupsthat  exposed in the tokens, but please have a look. If we can control exclusion though this label then we can go around the way services parse VOMS groups by creating different groups in IAM.  So there won't be the need of any further development. Thanks a lot Francesco!

**Comment by maarten-litmaath** [comment]
Hi all,
there are historical second-level groups for several countries participating in CMS, e.g. `/cms/itcms` for Italy, which presumably is deemed awkward now, e.g. because of namespace pollution?

If so, shouldn't the resources that care about those attributes have their VOMS configurations updated to start recognizing the new naming scheme?

That is, recognize `/cms/country/it` possibly in addition to `/cms/itcms` for the time being...

**Comment by stlammel** [comment]
Hallo Maarten,
   they would not go into the scope namespace, just appear in the wlcg.group list.
   Our plan is to keep the VOMS config at site unchanged, i.e. configure things on CRIC/IAM side so existing VOMS groups continue as on the legacy servers.
   The question is how best to do this and if we need an extra flag to make IAM groups invisible to VOMS if it should be more general and VOMS group name instead of a flag. If the wlcg.optional-group is reused/extended for VOMS, it does what we need.
   Thanks,
          cheers,  Stephan

**Comment by giacomini** [comment]
Just to confirm the requeste change: when a group in IAM is labeled as `wlcg.optional-group`, such group will appear neither in the token (which is already the case) nor in the VOMS AC (this is the change). Is that correct?

**Comment by Panos512** [comment]
Yes I believe that's good enough to cover the CMS use case! Thank you very much @giacomini
